drivers/include/lis2de12.h

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * @brief   LIS2DE12 FIFO data struct
+ */
+typedef union {
+    struct {
+        int16_t x;                      /**< X data in mili-g */
+        int16_t y;                      /**< Y data in mili-g */
+        int16_t z;                      /**< Z data in mili-g */
+    } axis;                             /**< named axis access */
+    int16_t data[3];                    /**< x, y, z data in mili-g */
+} lis2de12_fifo_data_t;
+
+/**
+ * @brief   LIS2DE12 FIFO modes
+ */
+typedef enum {
+    LIS2DE12_FIFO_MODE_BYPASS = 0,      /**< default mode, FIFO is bypassed */
+    LIS2DE12_FIFO_MODE_FIFOMODE,        /**< normal FIFO mode, stops if FIFO is full */
+    LIS2DE12_FIFO_MODE_STREAM,          /**< Stream mode, oldest values get overwritten */
+    LIS2DE12_FIFO_MODE_STREAMtoFIFO,    /**< Stream mode and on interrupt jumps to FIFO mode */
+} lis2de12_fifo_mode_t;
+
+/**
+ * @brief   LIS2DE12 FIFO config values
+ */
+typedef struct {
+    lis2de12_fifo_mode_t FIFO_mode; /**< set FIFO mode */
+    uint8_t FIFO_watermark:5;       /**< set the FIFO watermark level */
+    bool FIFO_set_INT2;             /**< sets the FIFO interrupt to INT2, otherwise INT1 */
+} lis2de12_fifo_t;
+
+
+/**
+ * @brief   Device descriptor for the driver
+ */
+typedef struct {
+    /** Device initialization parameters */
+    lis2de12_scale_t scale;         /**< sampling sensitivity used */
+    lis2de12_rate_t rate;           /**< sampling rate used */
+
+    const lis2de12_params_t *params;
+} lis2de12_t;
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    LIS2DE12_OK    =  0,            /**< everything was fine */
+    LIS2DE12_NOBUS = -1,            /**< bus interface error */
+    LIS2DE12_NODEV = -2,            /**< unable to talk to device */
+    LIS2DE12_NOINT = -3,            /**< wrong interrupt line (has to be LIS2DE12_INT1
+                                        or LIS2DE12_INT2) */
+    LIS2DE12_NODATA= -4,            /**< no data available */
+};
+
+/**
+ * @brief   Initialize the given device
+ *
+ * @param[inout] dev        Device descriptor of the driver
+ * @param[in]    params     Initialization parameters
+ *
+ * @return                  0 on success
+ */
+int lis2de12_init(lis2de12_t *dev, const lis2de12_params_t *params);
+
+/**
+ * @brief   Read acceleration data from the given device
+ *
+ * @param[in]  dev      device descriptor
+ * @param[out] data     acceleration data in mili-g
+ *
+ * @return  LIS2DE12_OK on success
+ * @return  LIS2DE12_NOBUS on bus error
+ */
+int lis2de12_read(const lis2de12_t *dev, lis2de12_fifo_data_t *data);
+
+
+int lis2de12_clear_data(const lis2de12_t *dev);
+
+int lis2de12_set_datarate(const lis2de12_t *dev, lis2de12_rate_t rate);
+uint16_t lis2de12_get_datarate(const lis2de12_t *dev);
+
+int lis2de12_set_scale(lis2de12_t *dev, lis2de12_scale_t scale);
+lis2de12_scale_t lis2de12_get_scale(lis2de12_t *dev);
+
+int lis2de12_set_fifo(const lis2de12_t *dev, const lis2de12_fifo_t *config);
```

Comment:
You are never doing anything with the fifo, you only read the first element with `lis2de12_read()`